tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py

```diff
@@ -96,3 +98,339 @@ def test_label_swirl(self, xp, scp):
+        return scp.ndimage.sum(image, label, 1)
+
+    @testing.for_dtypes([cupy.bool_, cupy.complex64, cupy.complex128])
+    def test_ndimage_sum_wrong_image_dtype(self, dtype):
+        image = cupy.arange(100).astype(dtype)
+        label = cupy.random.randint(1, 4, dtype=cupy.int32, size=100)
+        index = cupy.array([1, 2, 3])
+        with pytest.raises(TypeError):
+            cupyx.scipy.ndimage.sum(image, label, index)
+
+    def test_ndimage_sum_wrong_label_shape(self):
+        image = cupy.arange(100, dtype=cupy.int32)
+        label = cupy.random.randint(1, 3, dtype=cupy.int32, size=50)
+        index = cupy.array([1, 2, 3])
+        with pytest.raises(ValueError):
+            cupyx.scipy.ndimage.sum(image, label, index)
+
+    def test_ndimage_sum_wrong_image_type(self):
+        image = list(range(100))
+        label = cupy.random.randint(1, 3, dtype=cupy.int32, size=100)
+        index = cupy.array([1, 2, 3])
+        with pytest.raises(TypeError):
+            cupyx.scipy.ndimage.sum(image, label, index)
+
+    def test_ndimage_sum_wrong_label_type(self):
+        image = cupy.arange(100, dtype=cupy.int32)
+        label = numpy.random.randint(1, 3, dtype=numpy.int32, size=100)
+        index = cupy.array([1, 2, 3])
+        with pytest.raises(TypeError):
+            cupyx.scipy.ndimage.sum(image, label, index)
+
+    def test_ndimage_sum_wrong_index_type(self):
+        image = cupy.arange(100, dtype=cupy.int32)
+        label = cupy.random.randint(1, 3, dtype=cupy.int32, size=100)
+        index = [1, 2, 3]
+        with pytest.raises(TypeError):
+            cupyx.scipy.ndimage.sum(image, label, index)
+
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
+    def test_ndimage_sum_zero_values(self, xp, scp):
+        image = cupy.array([])
+        label = cupy.array([])
+        index = cupy.array([])
+        return cupyx.scipy.ndimage.sum(image, label, index)
+
+
+@testing.gpu
+@testing.with_requires('scipy')
+class TestNdimageVariance(unittest.TestCase):
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
+    def test_ndimage_variance(self, xp, scp, dtype):
+        image = xp.arange(100, dtype=dtype)
+        label = testing.shaped_random((100,), xp, dtype=xp.int32, scale=3)
+        index = xp.array([0, 1, 2])
+        return scp.ndimage.variance(image, label, index)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
+    def test_ndimage_variance_multi_dim(self, xp, scp, dtype):
+        image = xp.arange(512, dtype=dtype).reshape(8, 8, 8)
+        label = testing.shaped_random((8, 8, 8), xp, dtype=xp.int32, scale=3)
+        index = xp.array([0, 1, 2])
+        return scp.ndimage.variance(image, label, index)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True, no_float16=True)
```

Comment:
Can't we test `float16` inputs?

Reply:
This throws an error because of difference in accuracy.

Reply:
Thanks, it would be fine, or how about using smaller values for `float16` as:
```Python
if dtype == cupy.float16:
    image = testing.shaped_random((100,), xp, dtype=cupy.float16)
else:
    image = xp.arange(100, dtype=dtype)
```

Reply:
Thanks for the suggestion, unfortunately for variance and standard deviation, it fails. I tried to even reduce the number of elements for both arrange and random but it seems that both  lead to decimal differences at the 2nd decimal point. I dont think checking at 2nd decimal point is a good idea so perhaps we could make a test group for only float16 with looser checking?
Do you have any better suggestion for dealing with this ?

Reply:
I guess intermediate value is computed in `float16`. Would you make it done in `float64`, not just casting final output values?

Reply:
I did some testing and got the following results,
```python
from scipy.ndimage import mean as mean_n
import numpy as np
import cupy as cp

arr = np.asarray(np.arange(100)/2, dtype = np.float16)

print("simple no casted mean ",arr.mean())
print("scipy ndimage mean",mean_n(arr))
print("pre-casted numpy mean", arr.astype(np.float64).mean())
print("casted after mean", arr.mean().astype(np.float64))
print("cast in mean func",arr.mean(dtype=np.float64))

t = [i/2 for i in list(range(100))]
print("sanity check", sum(t)/len(t))

```

> simple no casted mean  24.75
> scipy ndimage mean 24.76
> pre-casted numpy mean 24.75
> casted after mean 24.75
> cast in mean func 24.75
> sanity check 24.75

The scipy mean failes to accurately calculate values for float16 for such small numbers even till the second decimal place, and since this mean value is propogated to variance and standard deviation, I dont think any test I write will pass unless and until I set a very low threshold. Please tell me how I should proceed.

Reply:
Thanks, I think your result makes sense. Hmm, then would you split the test methods for `float16` with supplying low threshold as `test_ndimage_only_input_fp16`? I'll take a look into scipy's implementation, or numpy's `bincount`.